src/scales/quantitative.js

```diff
@@ -284,3 +285,18 @@ function inferLogDomain(channels) {
   }
   return [1, 10];
 }
+
+// Certain marks have special behavior if a scale is collapsed, i.e. if the
+// domain is degenerate and represents only a single value such as [3, 3]; for
+// example, a rect will span the full extent of the chart along a collapsed
+// dimension (whereas a dot will simply be drawn in the center).
+export function isCollapsed(scale) {
```

Comment:
You moved this into quantitative.js, but this behavior isn’t limited to quantitative scales (although that is where it most often applies). See the ordinalBar example, which should also be considered collapsed if the domain only contains a single value.

Reply:
Can you clarify the ordinalBar example? I've tried several variations and all seem to give the same chart under both branches. For example `Plot.barY("A", {x: (d, i) => i, y: d => d}).plot()` gives a large rect (width x height).

Reply:
The example is with a point scale:

```js
Plot.plot({
  y: {
    type: "point",
    grid: true
  },
  marks: [
    Plot.barY([0, 0, 0], {x: (d, i) => i, y2: d => d}),
    Plot.ruleY([0])
  ]
})
```

This is admittedly a pretty pointless plot, but it does demonstrate the issue: in this PR, the _y_-domain is not considered collapsed because it is not a quantitative scale, even though its domain only has a single value [0].

<img width="624" alt="Screen Shot 2021-08-09 at 6 34 35 PM" src="https://user-images.githubusercontent.com/230541/128795125-96f3bdf2-3a5d-4cb9-b1af-5dd010e476c9.png">

Whereas in my PR you get:

<img width="618" alt="Screen Shot 2021-08-09 at 6 34 47 PM" src="https://user-images.githubusercontent.com/230541/128795120-bbb9aa35-3e75-4a88-babc-e003ede5d6e2.png">